server/components/auth-button/middleware.js

```diff
@@ -0,0 +1,63 @@
+/* @flow */
+import { clientErrorResponse, htmlResponse, allowFrame, defaultLogger, sdkMiddleware, getCSPNonce, type ExpressMiddleware } from '../../lib';
+import type { LoggerType, CacheType, InstanceLocationInformation } from '../../types';
+
+import { getPayPalAuthButtonsRenderScript } from './script';
+import { htmlTemplate } from './htmlTemplate';
+
+
+type AuthButtonMiddlewareOptions = {|
+    logger : LoggerType,
+    cache : CacheType,
+    getInstanceLocationInformation : () => InstanceLocationInformation,
+|};
+
+export function getAuthButtonMiddleware({ logger = defaultLogger, cache, getInstanceLocationInformation } : AuthButtonMiddlewareOptions) : ExpressMiddleware {
+    const locationInformation = getInstanceLocationInformation();
+    return sdkMiddleware({ logger, cache, locationInformation }, {
+        app: async ({ req, res, params, meta, logBuffer }) => {
+            const cspNonce = getCSPNonce(res);
+            const {
+                scopes,
+                buttonType,
+                responseType,
+                clientID,
+                returnurl,
+                customLabel,
+                locale,
+                style = {}
+            } = params;
+
+            if (!clientID) {
+                logger.info(req, 'smart_buttons_render');
+                return clientErrorResponse(res, 'Please provide a clientID query parameter');
+            }
+            logger.info(req, `auth_button clientID: ${clientID}`);
```

Comment:
I don't think we should put `clientID` in the log name because the cardinality on that would be extremely high. I would suggest:

```suggestion
            logger.info(req, 'auth_button', {clientID});
```

another suggestion would be to add this to the rootTxn like we do for the payment buttons: https://github.com/paypal/paypal-smart-payment-buttons/blob/main/server/components/buttons/instrumentation.js#L37